### DISCUSSION TITLE
Genre Diversity Custom Plugin

### DISCUSSION THREAD

**Discussion by alxtrnr** [beetbox/beets#5472] [Show and tell]
[genrediversity](https://github.com/alxtrnr/genrediversity) A Beets custom plugin that calculates and analyzes the genre diversity of your music library. The plugin provides insights into how diverse your music collection is, using metrics such as the Shannon and Simpson Diversity Indices.

**Comment by JOJ0** [comment]
as far as I understand it should also work when multiple genres on a track are simply joined by comma or semicolob?

**Reply by alxtrnr** [reply]
It's coded to work with comma separated genres.

**Reply by alxtrnr** [reply]
If using another seperator just find and edit the following line to suit your needs -

`primary_genres = [item.genre.split(',')[0] for item in genres if item.genre]`
